tools/src/main/scala/akka/contrib/persistence/mongodb/MigrateToSuffixedCollections.scala

```diff
@@ -0,0 +1,206 @@
+                } match { case ssTotOk => (ok + ssTotOk, tot + cnt) }
+            } match {
+              case (inserted, count) =>
+                logger.info(s"$inserted/$count records were inserted into '$collectionName' journal")
+                tempDbObjects.foldLeft(0L) {
+                  case (ok, tdbo) =>
+                    val query = pidQuery(tdbo.get("_id").toString)
+                    Try { journal.remove(query, journalWriteConcern) } map { r =>
+                      ok + r.getN
+                    } recover {
+                      case _: Throwable =>
+                        logger.warn(s"Errors occurred when trying to remove records, previously copied to '$collectionName' journal, from '${settings.JournalCollection}' journal")
+                        ok
+                    } getOrElse (ok)
+                } match {
+                  case removed =>
+                    logger.info(s"$removed/$count records, previously copied to '$collectionName' journal, were removed from '${settings.JournalCollection}' journal")
+                    if (removed < inserted) (done + removed, ignored) else (done + inserted, ignored)
+                }
+            }
+          } else {
+            tempDbObjects.foldLeft(0L) {
+              case (tot, tdbo) =>
+                val query = pidQuery(tdbo.get("_id").toString)
+                val cnt = journal.count(query).toLong
+                tot + cnt
+            } match { case ssTotIgnored => 
+                logger.info(s"$ssTotIgnored records were ignored and remain in '${settings.JournalCollection}' journal")
+                (done, ignored + ssTotIgnored)
+            }
+          }
+        }
+      } match { 
+        case (totalOk, totalIgnored) => 
+          logger.info(s"JOURNALS: $totalOk/$totalCount records were successfully transfered to suffixed collections")
+          if ( totalIgnored > 0) {
+            logger.info(s"JOURNALS: $totalIgnored/$totalCount records were ignored and remain in '${settings.JournalCollection}' journal")
+            if (totalIgnored + totalOk == totalCount)
+              logger.info(s"JOURNALS: $totalOk + $totalIgnored = $totalCount, all records were successfully handled")
+            else
+              logger.warn(s"JOURNALS: $totalOk + $totalIgnored does NOT equal $totalCount, check remaining records  in '${settings.JournalCollection}' journal")
+          }
+      }
+    }
+
+    // SNAPSHOTS //
+
+    // aggregate persistenceIds, contained in unique snaps, and store them in temporary collection
+    snaps.aggregate(
+      MongoDBObject("$group" -> MongoDBObject("_id" -> s"$$$PROCESSOR_ID")) ::
+        MongoDBObject("$out" -> temporaryCollectionName) ::
+        Nil).results
+
+    // for each retrieved persistenceId, insert corresponding records from unique snaps
+    // to appropriate (and newly created) suffixed snaps, then remove them from unique snaps
+    if (temporaryCollection.count() > 0) {
```

Comment:
Similar to the journal, this is a complicated section.  Can methods be pulled out here?